core/embed/rust/src/ui/layout_caesar/component/share_words.rs

```diff
@@ -149,13 +137,14 @@ impl<'a> Component for ShareWords<'a> {
 }
 
 impl<'a> Paginate for ShareWords<'a> {
-    fn page_count(&self) -> usize {
-        self.total_page_count()
+    fn pager(&self) -> Pager {
+        self.pager
     }
 
-    fn change_page(&mut self, active_page: usize) {
-        self.page_index = active_page;
-        self.scrollbar.change_page(active_page);
+    fn change_page(&mut self, active_page: u16) {
+        let to_page = active_page.min(self.pager.total() - 1);
+        self.pager.set_current(to_page);
+        self.scrollbar.change_page(to_page);
```

Comment:
`set_current` already implements saturation functionality
```suggestion
        self.pager.set_current(active_page);
        self.scrollbar.change_page(self.pager.current());
```

Reply:
https://github.com/trezor/trezor-firmware/pull/5425/commits/e9ae7b27742b063b36e5529004c7a45345c5052a

As for your other suggestion below, this PR changes the types to `u16` https://github.com/trezor/trezor-firmware/pull/5438